### PR TITLE
rustup, also first time clippy was used on itself, which led to a sma…

### PR DESCRIPTION
**Pull request by llogiq**
…ll refactoring

### CODE REVIEW

File: src/eta_reduction.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ impl LintPass for EtaPass {
 
     fn check_expr(&mut self, cx: &Context, expr: &Expr) {
         if let ExprClosure(_, ref decl, ref blk) = expr.node {
-            if blk.stmts.len() != 0 {
+            if !blk.stmts.is_empty() {
```
Comment (Manishearth) [inline src/eta_reduction.rs]: HA!

Comment (llogiq) [inline src/eta_reduction.rs]: Eating our own dog food... :smile:

**Comment by Manishearth** [comment]
Blocked on https://github.com/laumann/compiletest-rs/pull/20

**Comment by llogiq** [comment]
@Manishearth Don't we have an up to date nightly rustc on travis?

**Comment by Manishearth** [comment]
We do, compiletest isn't updated.

**Comment by llogiq** [comment]
Ah, I see.

File: src/eta_reduction.rs (hunk 1 above)
Comment (laumann) [inline src/eta_reduction.rs]: Sweet :smile:

**Comment by laumann** [comment]
@llogiq A new version of compiletest-rs is available

**Pull request merged**